solr/core/src/java/org/apache/solr/parser/SolrQueryParserBase.java

```diff
@@ -1159,8 +1160,11 @@ protected Query getFieldQuery(String field, List<String> queryTerms, boolean raw
             for (String queryTerm : queryTerms) {
               try {
                 subqs.add(ft.getFieldQuery(parser, sf, queryTerm));
-              } catch (Exception e) { // assumption: raw = false only when called from ExtendedDismaxQueryParser.getQuery()
-                // for edismax: ignore parsing failures
+              } catch (Exception e) {
+                // This happens when a field tries to parse a query term that has a type incompatible with the field
+                // e.g.
+                // a numerical field trying to parse a textual query term
+                subqs.add(new MatchNoDocsQuery(queryTerm + " is not compatible with " + field));
```

Comment:
Minor point, but I think it'd be fine to be less specific in the "reason" arg to the MatchNoDocsQuery. Not to micro-optimize, but the specificity is unlikely to _gain_ anything here practically speaking (right?). (A cursory check for existing MatchNoDocsQuery constructor invocations in Solr and Lucene shows that all seem to either invoke the no-arg ctor or pass in a static String)

Reply:
ok, no strong opinion here, let me revert to empty constructor

Reply:
Thanks, sure; or a static "reason" String, e.g. "field-incompatible clause should count as no match" -- in any case, point being to avoid dynamic creation of a String that'll likely be ignored anyway.